zc.recipe.egg_/src/zc/recipe/egg/egg.py

```diff
@@ -177,14 +185,21 @@ def working_set(
     links=(),
     index=None,
     allow_hosts=('*',),
+    cache_storage=_global_working_set_cache,
 ):
     """Helper function to build a working set.
 
     Return an instance of `pkg_resources.WorkingSet`.
 
-    Results are cached in the `_working_set_cache` dict at module level.
+    Results are cached in the mapping provided in the `cache_storage` argument.
+    This defaults to a global dict at this module. If `None` is passed then
+    caching is disabled.
```

Comment:
The reason is I think the `working_set()` function at module level is useful on its own. And if people choose to use it then it's nice to have caching enabled by default, since building working sets is slow. 

I think it's better to do this than putting the burden of setting up the cache storage in the callers.

For example, check out this code: https://github.com/plone/plone.recipe.zope2instance/blob/master/src/plone/recipe/zope2instance/__init__.py#L620

I think it would be nice to use the `working_set()` function here, instead of instantiating the `Egg` class.

Remember caching can be disable if needed, passing `cache_storage=None`.